dev/new-rule/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+node.js-CLI-based generation of new rule template files
+5 things will happen:
+1. a new rule file will be created
+2. a new test file will be created
+3. a new doc file will be created
+4. the rule will be appended to the list of rules in the index
+5. the doc will be appended to the list of docs in the doc index
+
+Example usage: 
+--------------
+Templates use the rule name `my-new-rule`, so this example
+creates a functioning/immediately testable rule
+# /path/to/ember-template-lint
+$ yarn new my-new-rule
+$ yarn test my-new-rule
+*/
+
+const fs = require('fs');
+const path = require('path');
+
+// CLI Argument: `my-new-rule` (name of new rule to add)
+let newRuleName = process.argv[2];
+
+let pathUtil = __dirname;
+let pathTemplates = path.join(pathUtil, 'templates');
+let pathRoot = path.join(pathUtil, '../..');
+
+// Create rule file
+let pathRuleTemplate = path.join(pathTemplates, 'rule.js');
+let pathRuleNew = path.join(pathRoot, 'lib', 'rules', newRuleName + '.js');
+function createNewRuleFile(newRuleName, pathRuleTemplate, pathRuleNew) {
+  let originalContent = fs.readFileSync(pathRuleTemplate, { encoding: 'utf8' });
+  let newRuleClassArray = newRuleName.split('-');
+  let newRuleClassArrayCaps = newRuleClassArray.map(element =>
+    element.replace(element.charAt(0), element.charAt(0).toUpperCase())
+  );
+  let newRuleClassName = newRuleClassArrayCaps.join('');
```

Comment:
Can you extract all this string manipulation into a separate function `toTitleCase(str)`?

Reply:
I see you added a helper `toTitleCase` but haven't used it yet?